### PR TITLE
embeddetection.py Rewrite

### PR DESCRIPTION
**Pull request by thewildofficial**
1. removed channel whitelist system($whitelist)
2.added add_embed command (to add a custom message per channel)
3.code cleanup

### CODE REVIEW

File: extensions/embeddetection.py [hunk 1]
```diff
@@ -1,44 +1,37 @@
...
-                        await message.author.send(embed=dm_message)
+        pic_ext = ['.jpg', '.png', '.jpeg']
+        if message.attachments != []:
+            for ext in pic_ext:
+                attach = str(message.attachments[0])
+                if attach.endswith(f"{ext}\'>"):
+                    for entry in self.channel_embed_information.all():
+                        if entry['channel-id'] == message.channel.id:
+                            embed_to_be_sent = Embed(title="Embed Detected!", color=0xff0000)
+                            embed_to_be_sent.add_field(name=f"Dearest {message.author.name},", value=entry['value'])
```
Comment (dscalvin97) [inline extensions/embeddetection.py:30]: The embed name being `"Hey TheWildOfficial"` might be a bit better than `"Dearest TheWildOfficial"`, but it's only my opinion here.
Please ask about this in the test server and see if if people are fine with the title as-is or would prefer something different, I've only provided a suggestion here.

File: extensions/embeddetection.py [hunk 2]
```diff
@@ -1,44 +1,37 @@
...
+                            MessageInChat = await message.channel.send(embed=embed_to_be_sent)
+                            await message.author.send(embed=embed_to_be_sent)
+                            await asyncio.sleep(90)
```
Comment (dscalvin97) [inline extensions/embeddetection.py:33]: Has this been tested to work when there are two separate embeds sent simultaneously? We don't want the coroutine execution to stop for 90 seconds before it starts looking for the next message with embeds.

Please test this out and revert with the results here, if this already works without stopping the coroutine execution, then feel free to resolve this conversation.

File: extensions/embeddetection.py [hunk 3]
```diff
@@ -1,44 +1,37 @@
...
+        pic_ext = ['.jpg', '.png', '.jpeg']
+        if message.attachments != []:
+            for ext in pic_ext:
+                attach = str(message.attachments[0])
+                if attach.endswith(f"{ext}\'>"):
+                    for entry in self.channel_embed_information.all():
+                        if entry['channel-id'] == message.channel.id:
+                            embed_to_be_sent = Embed(title="Embed Detected!", color=0xff0000)
+                            embed_to_be_sent.add_field(name=f"Dearest {message.author.name},", value=entry['value'])
+                            MessageInChat = await message.channel.send(embed=embed_to_be_sent)
```
Comment (dscalvin97) [inline extensions/embeddetection.py:31]: Please reformat the variable name from `MessageInChat` to `message-in-chat`.

Additionally, if possible, rename the variable to something that reflects it's use a bit better as `message-in-chat` is a bit ambiguous to it's usage.

File: extensions/embeddetection.py [hunk 4]
```diff
@@ -1,44 +1,37 @@
...
-
-        elif type == "remove":
-            remove_query = Query()
-            try:
-                self.channel_whitelist.remove(remove_query['channel-name'].search(channel.name))
-                await ctx.send(f"channel {channel.name} has been removed!")
-            except:
-                await ctx.send("Channel Not Found In Database")
+    async def add_embed(self, ctx, channel: discord.TextChannel, value):
+        # channel = channel in which embed should be sent,domain= the department in which the embed should be sent,value = message to be shown
```
Comment (dscalvin97) [inline extensions/embeddetection.py:16]: The comment seems to include arguments that are not in the function anymore, please update it.

File: extensions/embeddetection.py (hunk 2 above)
Comment (thewildofficial) [reply extensions/embeddetection.py:33]: yeah,that may be an issue,i will start some tests.

File: extensions/embeddetection.py (hunk 3 above)
Comment (thewildofficial) [reply extensions/embeddetection.py:31]: - characters are not allowed in python, i hope underscores will suffice.

Comment (thewildofficial) [reply extensions/embeddetection.py:31]: i ment dash chars

File: extensions/embeddetection.py [hunk 5]
```diff
@@ -1,44 +1,37 @@
...
+        self.channel_embed_information.insert({"channel-id": channel.id, "value": value})
```
[the remark this answers is not in the document]
Comment (thewildofficial) [reply extensions/embeddetection.py:17]: throwing a TypeError.

[the remark this answers is not in the document]
Comment (thewildofficial) [reply extensions/embeddetection.py:17]: your snippet throws the following error.
` TypeError: upsert() missing 1 required positional argument: 'cond'`

[the remark this answers is not in the document]
Comment (dscalvin97) [reply extensions/embeddetection.py:17]: Apologies for that, here's the fixed snippet.
```python
self.channel_embed_information.upsert({"channel-id": channel.id, "value": value}, where("channel-id") == channel.id)
```

File: extensions/embeddetection.py (hunk 2 above)
Comment (dscalvin97) [reply extensions/embeddetection.py:33]: Resolving as this is not an issue from testing 👍

File: extensions/embeddetection.py (hunk 3 above)
Comment (dscalvin97) [reply extensions/embeddetection.py:31]: Oh yes, sorry about that, I meant to use underscores so the name would be `message_in_chat`

Comment (thewildofficial) [reply extensions/embeddetection.py]: done.

File: extensions/embeddetection.py (hunk 4 above)
Comment (thewildofficial) [reply extensions/embeddetection.py]: fixed.

File: extensions/embeddetection.py (hunk 1 above)
Comment (thewildofficial) [reply extensions/embeddetection.py]: keeping this on hold for discussion.

File: extensions/embeddetection.py [hunk 6]
```diff
@@ -1,44 +1,37 @@
+import asyncio
+
 import discord
+from discord import Embed
 from discord.ext import commands
-from tinydb import Query, where
```
[the remark this answers is not in the document]
Comment (thewildofficial) [reply extensions/embeddetection.py:3]: done

File: extensions/embeddetection.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (thewildofficial) [reply extensions/embeddetection.py]: done

File: extensions/embeddetection.py [hunk 7]
```diff
@@ -1,44 +1,37 @@
...
-                        dm_message.add_field(name="Attention:",
-                                             value=f"Our System has detected that you have sent a message containing an {ext} file, if you are a 3D artist,please use CHANNEL or show your work to USER!")
-                        await message.author.send(embed=dm_message)
+        pic_ext = ['.jpg', '.png', '.jpeg']
+        if message.attachments != []:
+            for ext in pic_ext:
+                attach = str(message.attachments[0])
+                if attach.endswith(f"{ext}\'>"):
+                    for entry in self.channel_embed_information.all():
+                        if entry['channel-id'] == message.channel.id:
```
[the remark this answers is not in the document]
Comment (thewildofficial) [reply extensions/embeddetection.py]: done

File: extensions/embeddetection.py [hunk 8]
```diff
@@ -1,44 +1,37 @@
...
-                await ctx.send(f"{channel.name} Has Been Added To The DataBase!")
-
-        elif type == "remove":
-            remove_query = Query()
-            try:
-                self.channel_whitelist.remove(remove_query['channel-name'].search(channel.name))
-                await ctx.send(f"channel {channel.name} has been removed!")
-            except:
-                await ctx.send("Channel Not Found In Database")
+    async def add_embed(self, ctx, channel: discord.TextChannel, value):
```
[the remark this answers is not in the document]
Comment (thewildofficial) [reply extensions/embeddetection.py]: changed.

File: extensions/embeddetection.py (hunk 1 above)
Comment (dscalvin97) [reply extensions/embeddetection.py]: No issues, we can update it later if admin needs it, resolving comment.

**Pull request merged**